### PR TITLE
fix: tests

### PR DESCRIPTION
**Pull request by BiswaViraj**
### Screenshots

- fixes cypress tests
<details>
<summary><strong>Expand for optional sections</strong></summary>

### Special notes for your reviewer

</details>

### CODE REVIEW

File: apps/web/src/pages/templates/components/StepEditorSidebar.tsx [hunk 1]
```diff
@@ -62,6 +72,7 @@ export const StepEditorSidebar = ({ children }: { children: ReactNode }) => {
         navigate(basePath);
       }}
       data-test-id="step-editor-sidebar"
+      onSubmit={handleSubmit(onSubmitHandler, onInvalid)}
```
Comment (antonjoel82) [inline apps/web/src/pages/templates/components/StepEditorSidebar.tsx:75]: ❓ **question:** What happened here -- did someone make changes that caused this or is this related to our feature work?‏

File: apps/web/cypress/tests/settings/user-profile.spec.ts
Comment (antonjoel82) [inline apps/web/cypress/tests/settings/user-profile.spec.ts:1]: ☑ **chore:** I believe we can reset this file since I don't think `@` is necessary in the password since we have `!`‏

File: apps/web/cypress/tests/organization-page.spec.ts [hunk 2]
```diff
@@ -1,5 +1,6 @@
 describe('Organization Page', function () {
   beforeEach(function () {
+    cy.mockFeatureFlags({ IS_INFORMATION_ARCHITECTURE_ENABLED: true });
```
Comment (antonjoel82) [inline apps/web/cypress/tests/organization-page.spec.ts:3]: 👏 **praise:** Good catch!‏

File: apps/web/src/pages/templates/components/StepEditorSidebar.tsx (hunk 1 above)
Comment (BiswaViraj) [inline apps/web/src/pages/templates/components/StepEditorSidebar.tsx:75]: Due to this, it was not submitting the form on the editor and hence the tests were failing on update workflow

File: apps/web/cypress/tests/organization-brand.spec.ts [hunk 3]
```diff
@@ -1,5 +1,6 @@
 describe('Organization Brand Screen', function () {
   beforeEach(function () {
+    cy.mockFeatureFlags({ IS_INFORMATION_ARCHITECTURE_ENABLED: true });
```
Comment (antonjoel82) [inline apps/web/cypress/tests/organization-brand.spec.ts:3]: ☑ **chore:** While this is good for one case, we also need the tests to confirm we haven't broken anything in the original flow‏

File: apps/web/cypress/tests/layout/main-nav.spec.ts [hunk 4]
```diff
@@ -85,33 +85,25 @@ describe('Main Nav (Sidebar)', () => {
       cy.getByTestId('side-nav-settings-team-link').should('exist');
       cy.getByTestId('side-nav-settings-branding-link').should('exist');
       cy.getByTestId('side-nav-settings-billing-link').should('exist');
-
-      // TODO: remove these lines when we implement toggles for different envs
-      cy.getByTestId('side-nav-settings-api-keys').should('exist');
-      cy.getByTestId('side-nav-settings-inbound-webhook').should('exist');
```
Comment (antonjoel82) [inline apps/web/cypress/tests/layout/main-nav.spec.ts:91]: ☑ **chore:** Please - reinstate these as they verify the buttons are there in their current form‏

**Review by antonjoel82 — APPROVED** [review]
Thanks for the good work on this and my apologies for the double-work! Please address the chores before merging, but otherwise we should be good!

File: apps/web/src/pages/templates/components/StepEditorSidebar.tsx (hunk 1 above)
Comment (BiswaViraj) [reply apps/web/src/pages/templates/components/StepEditorSidebar.tsx:75]: Yes, the Sidebar component was refactored and the form component inside the Sidebar is conditionally rendered now only when onSubmit is passed. Previously, the form used to render always.

File: apps/web/cypress/tests/organization-brand.spec.ts (hunk 3 above)
Comment (BiswaViraj) [reply apps/web/cypress/tests/organization-brand.spec.ts:3]: The orginal flow has a different spec file `branding.spec.ts`

File: apps/web/cypress/tests/layout/main-nav.spec.ts (hunk 4 above)
Comment (BiswaViraj) [reply apps/web/cypress/tests/layout/main-nav.spec.ts:91]: Good catch. Messed up in the rebase

File: apps/web/cypress/tests/organization-brand.spec.ts (hunk 3 above)
Comment (antonjoel82) [reply apps/web/cypress/tests/organization-brand.spec.ts:3]: Oh whoops sorry -- carry on!

File: apps/web/src/pages/templates/components/StepEditorSidebar.tsx (hunk 1 above)
Comment (antonjoel82) [reply apps/web/src/pages/templates/components/StepEditorSidebar.tsx:75]: Ah sorry this is my fault! But also points out the pattern of components that do too much and implicitly have extra behavior
